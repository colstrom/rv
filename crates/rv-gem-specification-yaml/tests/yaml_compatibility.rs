use rv_gem_specification_yaml::parse;
use std::fs;

fn load_fixture(name: &str) -> String {
    let fixture_path = format!("tests/fixtures/{name}.yaml");
    fs::read_to_string(&fixture_path)
        .unwrap_or_else(|_| panic!("Failed to read fixture: {fixture_path}"))
}

#[test]
fn test_parse_simple_specification() {
    let yaml_content = load_fixture("simple_spec");
    let result = parse(&yaml_content);

    match result {
        Ok(spec) => {
            insta::assert_debug_snapshot!("simple_spec_parsed", spec);
        }
        Err(e) => {
            panic!("Failed to parse simple specification: {e}");
        }
    }
}

#[test]
fn test_parse_complex_specification() {
    let yaml_content = load_fixture("complex_spec");
    let result = parse(&yaml_content);

    match result {
        Ok(spec) => {
            insta::assert_debug_snapshot!("complex_spec_parsed", spec);
        }
        Err(e) => {
            panic!("Failed to parse complex specification: {e}");
        }
    }
}

#[test]
fn test_parse_version_constraints_specification() {
    let yaml_content = load_fixture("version_constraints_spec");
    let result = parse(&yaml_content);

    match result {
        Ok(spec) => {
            insta::assert_debug_snapshot!("version_constraints_spec_parsed", spec);
        }
        Err(e) => {
            panic!("Failed to parse version constraints specification: {e}");
        }
    }
}

#[test]
fn test_parse_minimal_specification() {
    let yaml_content = load_fixture("minimal_spec");
    let result = parse(&yaml_content);

    match result {
        Ok(spec) => {
            insta::assert_debug_snapshot!("minimal_spec_parsed", spec);
        }
        Err(e) => {
            panic!("Failed to parse minimal specification: {e}");
        }
    }
}

#[test]
fn test_parse_prerelease_specification() {
    let yaml_content = load_fixture("prerelease_spec");
    let result = parse(&yaml_content);

    match result {
        Ok(spec) => {
            insta::assert_debug_snapshot!("prerelease_spec_parsed", spec);
        }
        Err(e) => {
            panic!("Failed to parse prerelease specification: {e}");
        }
    }
}

#[test]
fn test_parse_licensed_specification() {
    let yaml_content = load_fixture("licensed_spec");
    let result = parse(&yaml_content);

    match result {
        Ok(spec) => {
            insta::assert_debug_snapshot!("licensed_spec_parsed", spec);
        }
        Err(e) => {
            panic!("Failed to parse licensed specification: {e}");
        }
    }
}

#[test]
fn test_parse_edge_case_specification() {
    let yaml_content = load_fixture("edge_case_spec");
    let result = parse(&yaml_content);

    match result {
        Ok(spec) => {
            insta::assert_debug_snapshot!("edge_case_spec_parsed", spec);
        }
        Err(e) => {
            panic!("Failed to parse edge case specification: {e}");
        }
    }
}
