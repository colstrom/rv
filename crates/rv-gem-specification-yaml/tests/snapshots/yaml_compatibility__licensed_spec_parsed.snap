---
source: crates/rv-gem-specification-yaml/tests/yaml_compatibility.rs
expression: spec
---
Specification {
    name: "licensed-gem",
    version: Version {
        version: "1.5.2",
        segments: [
            Number(
                1,
            ),
            Number(
                5,
            ),
            Number(
                2,
            ),
        ],
    },
    summary: "A gem with extensive metadata",
    require_paths: [
        "lib",
        "ext",
    ],
    rubygems_version: "3.4.20",
    specification_version: 4,
    date: "",
    authors: [
        "License Developer",
        "Meta Author",
    ],
    email: [
        "dev@example.com",
        "meta@example.com",
    ],
    homepage: Some(
        "https://github.com/example/licensed-gem",
    ),
    description: Some(
        "This gem demonstrates comprehensive metadata usage including licensing, URLs, and custom fields.",
    ),
    licenses: [
        "MIT",
        "Apache-2.0",
    ],
    files: [
        "lib/licensed.rb",
        "README.md",
        "LICENSE",
    ],
    executables: [
        "licensed-cli",
    ],
    extensions: [],
    dependencies: [],
    metadata: {
        "bug_tracker_uri": "https://github.com/example/licensed-gem/issues",
        "changelog_uri": "https://github.com/example/licensed-gem/blob/main/CHANGELOG.md",
        "documentation_uri": "https://rubydoc.info/gems/licensed-gem",
        "funding_uri": "https://github.com/sponsors/example",
        "mailing_list_uri": "https://groups.google.com/forum/#\\!forum/licensed-gem",
        "source_code_uri": "https://github.com/example/licensed-gem",
        "wiki_uri": "https://github.com/example/licensed-gem/wiki",
        "custom_field": "custom_value",
    },
    platform: Ruby,
    bindir: "bin",
    post_install_message: None,
    requirements: [
        "libssl >= 1.1",
        "cmake >= 3.10",
    ],
    required_ruby_version: Requirement {
        constraints: [
            VersionConstraint {
                operator: GreaterEqual,
                version: Version {
                    version: "2.7.0",
                    segments: [
                        Number(
                            2,
                        ),
                        Number(
                            7,
                        ),
                        Number(
                            0,
                        ),
                    ],
                },
            },
        ],
    },
    required_rubygems_version: Requirement {
        constraints: [
            VersionConstraint {
                operator: GreaterEqual,
                version: Version {
                    version: "3.0.0",
                    segments: [
                        Number(
                            3,
                        ),
                        Number(
                            0,
                        ),
                        Number(
                            0,
                        ),
                    ],
                },
            },
        ],
    },
    test_files: [],
    extra_rdoc_files: [],
    rdoc_options: [],
    cert_chain: [],
    signing_key: None,
    autorequire: None,
    installed_by_version: None,
}
